tiledb/sm/c_api/tiledb.h

```diff
@@ -4690,6 +4776,622 @@ TILEDB_EXPORT int32_t tiledb_query_condition_combine(
+ * @param dim_idx The index of the dimension to add the range to.
+ * @param start The range start.
+ * @param end The range end.
+ * @param stride The range stride.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ *
+ * @note The stride is currently unsupported. Use 0/NULL/nullptr as the
+ *     stride argument.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_add_range(
+    tiledb_ctx_t* ctx,
+    tiledb_subarray_t* subarray,
+    uint32_t dim_idx,
+    const void* start,
+    const void* end,
+    const void* stride);
+
+/**
+ * Adds a 1D range along a subarray dimension name, which is in the form
+ * (start, end, stride). The datatype of the range components
+ * must be the same as the type of the domain of the array in the query.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * char* dim_name = "rows";
+ * int64_t start = 10;
+ * int64_t end = 20;
+ * tiledb_subarray_add_range_by_name(ctx, subarray, dim_name, &start, &end,
+ * nullptr);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param subarray The subarray to add the range to.
+ * @param dim_name The name of the dimension to add the range to.
+ * @param start The range start.
+ * @param end The range end.
+ * @param stride The range stride.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ *
+ * @note The stride is currently unsupported. Use 0/NULL/nullptr as the
+ *     stride argument.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_add_range_by_name(
+    tiledb_ctx_t* ctx,
+    tiledb_subarray_t* subarray,
+    const char* dim_name,
+    const void* start,
+    const void* end,
+    const void* stride);
+
+/**
+ * Adds a 1D variable-sized range along a subarray dimension index, which is in
+ * the form (start, end). Applicable only to variable-sized dimensions.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint32_t dim_idx = 2;
+ * char start[] = "a";
+ * char end[] = "bb";
+ * tiledb_subarray_add_range_var(ctx, subarray, dim_idx, start, 1, end, 2);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param subarray The subarray to add the range to.
+ * @param dim_idx The index of the dimension to add the range to.
+ * @param start The range start.
+ * @param start_size The size of the range start in bytes.
+ * @param end The range end.
+ * @param end_size The size of the range end in bytes.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_add_range_var(
+    tiledb_ctx_t* ctx,
+    tiledb_subarray_t* subarray,
+    uint32_t dim_idx,
+    const void* start,
+    uint64_t start_size,
+    const void* end,
+    uint64_t end_size);
+
+/**
+ * Adds a 1D variable-sized range along a subarray dimension name, which is in
+ * the form (start, end). Applicable only to variable-sized dimensions.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * char* dim_name = "rows";
+ * char start[] = "a";
+ * char end[] = "bb";
+ * tiledb_subarray_add_range_var_by_name(ctx, subarray, dim_name, start, 1, end,
```

Comment:
We could format these lines a bit better by forcing a new line at a better place, e.g., 

```c
tiledb_subarray_add_range_var_by_name(
    ctx, subarray, dim_name, start, 1, end, 2);
```

We can do this in all other similar places as well.

Reply:
Done.